frontend/src/pages/TaskMapPage.tsx

```diff
@@ -0,0 +1,191 @@
+import { FC, useState } from 'react';
+import { shallowEqual, useSelector, useDispatch } from 'react-redux';
+import ReactFlow, { Handle } from 'react-flow-renderer';
+import classNames from 'classnames';
+import { RootState } from '../redux/types';
+import {
+  allTasksSelector,
+  chosenRoadmapSelector,
+  taskSelector,
+} from '../redux/roadmaps/selectors';
+import { Roadmap, Task } from '../redux/roadmaps/types';
+import { StoreDispatchType } from '../redux';
+import { TaskRatingsText } from '../components/TaskRatingsText';
+import { groupTaskRelations } from '../utils/TaskRelationUtils';
+import { TaskOverview } from '../components/TaskOverview';
+
+import css from './TaskMapPage.module.scss';
+import { roadmapsActions } from '../redux/roadmaps';
+
+const classes = classNames.bind(css);
+
+// Node positions require special Position-enum in typescript
+enum Position {
+  Left = 'left',
+  Top = 'top',
+  Right = 'right',
+  Bottom = 'bottom',
+}
+
+// To do: Lisää Taskille mahdollisuus olla completed, jolloin sen tekstit on vihreet
+const SingleTask: FC<{
+  taskId: number;
+  selected?: boolean;
+  setSelectedTask?: any;
+}> = ({ taskId, setSelectedTask, selected }) => {
+  const task = useSelector<RootState, Task | undefined>(
+    taskSelector(taskId),
+    shallowEqual,
+  );
+  if (task)
+    return (
```

Comment:
```suggestion
  if (!task) return null;
  return (
```
I would invert the condition here